src/map/npc.cpp

```diff
@@ -5294,20 +5294,29 @@ static const char* npc_parse_mob(char* w1, char* w2, char* w3, char* w4, const c
 		}
 		mob.xs = 0;
 	}
+	else if (mob.xs == 0) {
+		// Both 0 and 1 result in fixed x-coordinate
+		mob.xs = 1;
```

Comment:
This is wrong, we need to eliminate 0 here. We substract 1 later and don't want it to go negative because that would mean map-wide. Maybe I need to improve the comment.

Reply:
See comment above.